src/api/schema/metrics/mod.rs

```diff
@@ -112,6 +116,98 @@ impl MetricsSubscription {
+    ) -> impl Stream<Item = i64> {
+        counter_throughput(interval, &|m| m.name() == "events_in_total")
+            .map(|(_, throughput)| throughput as i64)
+    }
+
+    /// Component event inputting throughput metrics over `interval`.
+    async fn component_events_in_throughputs(
+        &self,
+        #[graphql(default = 1000, validator(IntRange(min = "10", max = "60_000")))] interval: i32,
+    ) -> impl Stream<Item = Vec<ComponentEventsInThroughput>> {
+        component_counter_throughputs(interval, &|m| m.name() == "events_in_total").map(|m| {
+            m.into_iter()
+                .map(|(m, throughput)| {
+                    ComponentEventsInThroughput::new(
+                        m.tag_value("component_name").unwrap(),
+                        throughput as i64,
+                    )
+                })
+                .collect()
+        })
+    }
+
+    /// Component event inputting metrics over `interval`.
+    async fn component_events_in_totals(
+        &self,
+        #[graphql(default = 1000, validator(IntRange(min = "10", max = "60_000")))] interval: i32,
+    ) -> impl Stream<Item = Vec<ComponentEventsInTotal>> {
+        component_counter_metrics(interval, &|m| m.name() == "events_in_total")
+            .map(|m| m.into_iter().map(ComponentEventsInTotal::new).collect())
+    }
+
+    /// Event outputting metrics.
+    async fn events_out_total(
+        &self,
+        #[graphql(default = 1000, validator(IntRange(min = "10", max = "60_000")))] interval: i32,
+    ) -> impl Stream<Item = EventsOutTotal> {
+        get_metrics(interval).filter_map(|m| match m.name() {
+            "events_out_total" => Some(EventsOutTotal::new(m)),
+            _ => None,
+        })
+    }
+
+    /// Event outputting throughput sampled over the provided millisecond `interval`.
+    async fn events_out_throughput(
+        &self,
+        #[graphql(default = 1000, validator(IntRange(min = "10", max = "60_000")))] interval: i32,
+    ) -> impl Stream<Item = i64> {
+        counter_throughput(interval, &|m| m.name() == "events_out_total")
+            .map(|(_, throughput)| throughput as i64)
+    }
+
+    /// Component event outputting throughput metrics over `interval`.
+    async fn component_events_out_throughputs(
+        &self,
+        #[graphql(default = 1000, validator(IntRange(min = "10", max = "60_000")))] interval: i32,
+    ) -> impl Stream<Item = Vec<ComponentEventsOutThroughput>> {
+        component_counter_throughputs(interval, &|m| m.name() == "events_out_total").map(|m| {
+            m.into_iter()
+                .map(|(m, throughput)| {
+                    ComponentEventsOutThroughput::new(
+                        m.tag_value("component_name").unwrap(),
+                        throughput as i64,
+                    )
+                })
+                .collect()
+        })
+    }
+
+    /// Component event outputting metrics over `interval`.
```

Comment:
```suggestion
    /// Total outgoing component event metrics over `interval`
```

Noting that we generally haven't added a period suffix to GraphQL doc comments so far. Removing for consistency.